### PR TITLE
Extend Leaflet Util namespace

### PR DESCRIPTION
**Pull request by DenisCarriere**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 

http://leafletjs.com/reference-1.0.3.html#util
https://github.com/Leaflet/Leaflet/blob/master/src/core/Util.js

### CODE REVIEW

File: leaflet/index.d.ts [hunk 1]
```diff
@@ -1524,6 +1524,26 @@ declare namespace L {
...
     }
 
+    export namespace Util {
+        export function extend(dest: {}, src?: {}): {};
+        export function create(proto: {}, properties?: {}): {};
+        export function bind(fn: () => void, ...obj: Array<{}>): () => void;
+        export function stamp(obj: {}): number;
+        export function throttle(fn: () => void, time: number, context: {}): () => void;
+        export function wrapNum(num: number, range: number[], includeMax?: boolean): number;
+        export function falseFn(): () => void;
```
Comment (alejo90) [inline leaflet/index.d.ts]: According to the docs, this returns a function that always returns `false`, so the signature for the return type can be `() => boolean` or maybe even `() => false`

File: leaflet/index.d.ts [hunk 2]
```diff
@@ -1524,6 +1524,26 @@ declare namespace L {
...
+        export function formatNum(num: number, digits?: number): number;
+        export function trim(str: string): string;
+        export function splitWords(str: string): string[];
+        export function setOptions(obj: {}, options: {}): {};
+        export function getParamString(obj: {}, existingUrl?: string, uppercase?: boolean): string;
+        export function template(str: string, data: {}): string;
+        export function isArray(obj: {}): boolean;
+        export function indexOf(array: any[], el: {}): number;
+        export function requestAnimFrame(fn: () => void, context?: {}, immediate?: boolean): number;
+        export function cancelAnimFrame(id: number): undefined;
```
Comment (alejo90) [inline leaflet/index.d.ts]: The return type can be `void`

File: leaflet/index.d.ts [hunk 3]
```diff
@@ -1524,6 +1524,26 @@ declare namespace L {
         export const svg: boolean;
     }
 
+    export namespace Util {
+        export function extend(dest: {}, src?: {}): {};
```
Comment (alejo90) [inline leaflet/index.d.ts]: I suggest changing `{}` to `any` since `{}` is pretty unusual in the definitions I've looked at so far.

**Review by alejo90 — CHANGES_REQUESTED** [review]
Thanks, @DenisCarriere. There are a couple improvements we can make.

File: leaflet/index.d.ts [hunk 4]
```diff
@@ -1524,6 +1524,26 @@ declare namespace L {
...
+        export function splitWords(str: string): string[];
+        export function setOptions(obj: {}, options: {}): {};
+        export function getParamString(obj: {}, existingUrl?: string, uppercase?: boolean): string;
+        export function template(str: string, data: {}): string;
+        export function isArray(obj: {}): boolean;
+        export function indexOf(array: any[], el: {}): number;
+        export function requestAnimFrame(fn: () => void, context?: {}, immediate?: boolean): number;
+        export function cancelAnimFrame(id: number): undefined;
+        export const emptyImageUrl: string;
+    }
```
Comment (alejo90) [inline leaflet/index.d.ts]: The `lastId` property is missing.

**Comment by DenisCarriere** [comment]
@alejo90 Thanks for the feedback! I'll make those changes

File: leaflet/index.d.ts (hunk 3 above)
Comment (DenisCarriere) [inline leaflet/index.d.ts]: I used `{}` based on the Typescript linter.

```
1539:45  forbidden-types  Avoid using the Object type. Did you mean `any` or `{}`?
```
However I do agree it looks kinda weird :) I'll use `any` instead

File: leaflet/index.d.ts (hunk 4 above)
Comment (DenisCarriere) [inline leaflet/index.d.ts]: Doesn't look like `lastId` is exported or modified when using `stamp`.

https://github.com/Leaflet/Leaflet/blob/master/src/core/Util.js#L48-L57

Going to include it and make a PR to leaflet to export that variable

Comment (DenisCarriere) [inline leaflet/index.d.ts]: https://github.com/Leaflet/Leaflet/pull/5349

**Comment by alejo90** [comment]
👍

**Comment by DenisCarriere** [comment]
Looks like this could be merged, author gave a :+1: and all the suggestions have been reviewed and fixed.

CC: @zhengbli

**Comment by zhengbli** [comment]
👍 Thanks @DenisCarriere !

**Pull request merged**